src/java.compiler/share/classes/javax/lang/model/util/Types.java

```diff
@@ -35,6 +35,10 @@
 /**
  * Utility methods for operating on types.
  *
+ * Where a method returns a type mirror or a collection of type
+ * mirrors, any type mirrors represent types with no type annotations,
+ * unless otherwise indicated.
```

Comment:
I'm seeing `erasure` preserve type annotations on the array type (but not its component) after this change: `erasure(@A int @B []) = int @B []`. (Here's the [repro](https://gist.github.com/cushon/e0a59cc291c3b83204b9139addb01340).)

Is that deliberate, or is this still work in progress?

Reply:
(This comment is obsolete, it was on a much earlier version of the PR, and these issues have been addressed. The latest version correct handles removing annotations from contained types. There's some discussion about the issues with the earlier approach in https://github.com/openjdk/jdk/pull/8984#issuecomment-1806211798)

Reply:
> (This comment is obsolete, it was on a much earlier version of the PR, and these issues have been addressed. The latest version correct handles removing annotations from contained types. There's some discussion about the issues with the earlier approach in [#8984 (comment)](https://github.com/openjdk/jdk/pull/8984#issuecomment-1806211798))

Thanks, I guess I was confused by the different forms of the PR.